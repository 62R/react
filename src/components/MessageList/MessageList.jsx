import { Message } from '../Message';
import style from './MessageList.module.scss';

export const MessageList = ({ list }) => (
  <div className={style.messageList}>
    {list.map((item, idx) => (
      <Message key={idx} msg={item} />
    ))}
  </div>
);
