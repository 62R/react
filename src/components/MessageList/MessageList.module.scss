.messageList {
    align-self: start;
    margin: 20px 0;
}