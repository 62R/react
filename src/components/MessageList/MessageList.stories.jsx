import React from 'react';
import { MessageList } from './MessageList';

export default {
  title: 'MyApp/MessageList',
  component: MessageList,
};

const Template = (args) => <MessageList {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  list: [
    { author: 'John', msg: 'Some message' },
    { author: 'Bot', msg: 'Another message' },
  ],
};
