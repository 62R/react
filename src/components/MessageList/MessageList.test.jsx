import { MessageList } from './MessageList';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

describe('MessegeList', () => {
  it('render component snapshot', () => {
    const list = [
      { author: 'author', msg: 'msg1' },
      { author: 'author', msg: 'msg2' },
    ];
    const { asFragment } = render(<MessageList list={list} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
