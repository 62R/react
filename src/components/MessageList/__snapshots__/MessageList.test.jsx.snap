// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessegeList render component snapshot 1`] = `
<DocumentFragment>
  <div
    class="messageList"
  >
    <div
      class="message "
    >
      <p
        class="message__author"
      >
        author
      </p>
      <p
        class="message__message"
      >
        msg1
      </p>
    </div>
    <div
      class="message "
    >
      <p
        class="message__author"
      >
        author
      </p>
      <p
        class="message__message"
      >
        msg2
      </p>
    </div>
  </div>
</DocumentFragment>
`;
